src/datasets/utils/file_utils.py

```diff
@@ -229,17 +229,28 @@ class DownloadConfig:
     """
 
     cache_dir: Optional[Union[str, Path]] = None
-    force_download: bool = False
+    force_download: Optional[bool] = None  # default False
+    _is_force_download_default: bool = False
     resume_download: bool = False
     local_files_only: bool = False
     proxies: Optional[Dict] = None
     user_agent: Optional[str] = None
     extract_compressed_file: bool = False
     force_extract: bool = False
-    use_etag: bool = True
+    use_etag: Optional[bool] = None  # default True
+    _is_use_etag_default: bool = False
     num_proc: Optional[int] = None
     max_retries: int = 1
     use_auth_token: Optional[Union[str, bool]] = None
+    splits: Optional[list] = None
```

Comment:
The DownloadConfig class was used until now as a the class that defines the parameters that we pass to the `cached_path` function, and it had no logic related to the DownloadManager or to datasets specifically.
Therefore I'm not sure this is best place to put this argument. Let me know what you think.

Maybe this could be an argument of the DownloadManager itself.

Reply:
I have not a clear opinion on this yet (I'm going to think about it), but I can tell you that for readability, a data class called `DownloadConfig` should be clearly used by a `DownloadManager` class. Analogously, `cached_path` should use something like `CachedPathConfig`.

In my opinion I find quite sensible to include a download configuration setting (whether to download all or only some of the files) in a `DownloadConfig` object. And in order to avoid passing lots of parameters to `load_dataset`, it makes sense to pass the parameters related to `download` stuff in a `DownloadConfig` class.

As a side note, I am planning to refactor also `cached_path` because it contains too many different functionalities. For the moment, I have extracted all Extract functionalities. But I am planning further refactoring...

Reply:
The separation between the two configs makes sense to me, and I totally agree with you with the naming.
Maybe we could have `DownloadConfig` as a subclass of `CachedPathConfig` or something like that (or another relation between the two).

Regarding cached_path, note that this is a function that we have in common with the other libraries (transformers and huggingface_hub) so it could be worth discussing with the other maintainers about the changes we want to do.